test/DollarMintingCalculator.test.ts

```diff
@@ -0,0 +1,141 @@
+    // list of accounts
+    [admin] = await ethers.getSigners();
+    // deploy manager
+    const UADMgr = await ethers.getContractFactory(
+      "UbiquityAlgorithmicDollarManager"
+    );
+    manager = (await UADMgr.deploy(
+      await admin.getAddress()
+    )) as UbiquityAlgorithmicDollarManager;
+
+    // setup the oracle
+    metaPoolAddr = await manager.stableSwapMetaPoolAddress();
+
+    // set Dollar Minting Calculator
+    const dollarMintingCalculatorFactory = await ethers.getContractFactory(
+      "DollarMintingCalculator"
+    );
+    dollarMintingCalculator = (await dollarMintingCalculatorFactory.deploy(
+      manager.address
+    )) as DollarMintingCalculator;
+    await manager.setDollarCalculatorAddress(dollarMintingCalculator.address);
+  });
+  it("getDollarsToMint should revert with price less than 1$", async () => {
+    const totSupply = ethers.utils.parseEther("10000");
+    const uadPrice = ethers.utils.parseEther("0.9");
+    const price3CRV = ethers.utils.parseEther("1");
+    await setup(totSupply, uadPrice, price3CRV);
+    await expect(dollarMintingCalculator.getDollarsToMint()).to.be.reverted;
+  });
+  it("getDollarsToMint should return zero with price equal to 1$", async () => {
+    const totSupply = ethers.utils.parseEther("10000");
+    const uadPrice = ethers.utils.parseEther("1");
+    await setup(totSupply, uadPrice, uadPrice);
+    const toMint = await dollarMintingCalculator.getDollarsToMint();
+    expect(toMint).to.equal(0);
+  });
+  it("getDollarsToMint should return 10% of total supply with price equal to 1.1$", async () => {
+    const totSupply = ethers.utils.parseEther("10000");
+    const uadPrice = ethers.utils.parseEther("1.1");
+    await setup(totSupply, uadPrice, uadPrice);
+    const toMint = await dollarMintingCalculator.getDollarsToMint();
+    expect(toMint).to.equal(ethers.utils.parseEther("1000"));
+  });
+  it("getDollarsToMint should work with price above 1$", async () => {
+    const totSupply = ethers.utils.parseEther("10000");
+    const uadPrice = ethers.utils.parseEther("1.054678911145683254");
+    await setup(totSupply, uadPrice, uadPrice);
+    const toMint = await dollarMintingCalculator.getDollarsToMint();
+    const calculatedToMint = calcDollarsToMint(
+      new Big(totSupply.toString()),
+      new Big(uadPrice.toString())
+    );
+    expect(toMint).to.equal(calculatedToMint);
+  });
+  it("getDollarsToMint lose precision if supply is too large", async () => {
+    const totSupply = ethers.utils.parseEther("999999999999999999");
+
+    const uadPrice = ethers.utils.parseEther("1.054678911145683254");
+    await setup(totSupply, uadPrice, uadPrice);
+    // check tfor overflow revert
+    const toMint = await dollarMintingCalculator.getDollarsToMint();
+    const calculatedToMint = calcDollarsToMint(
+      new Big(totSupply.toString()),
+      new Big(uadPrice.toString())
+    );
+    expect(toMint).not.to.equal(calculatedToMint);
```

Comment:
should test with some precision p =>   delta < 10**-p

(as done for tests of durationMultiplier in BondingShare branch)